src/metabase/driver/postgres/actions.clj

```diff
@@ -3,52 +3,58 @@
   (:require
    [clojure.java.jdbc :as jdbc]
    [metabase.driver.sql-jdbc.actions :as sql-jdbc.actions]
+   [metabase.driver.sql-jdbc.connection :as sql-jdbc.conn]
+   [metabase.util :as u]
    [metabase.util.i18n :refer [tru]]))
 
-(defn- constraint->columns [conn constraint-name]
-  (->> ["select column_name from information_schema.constraint_column_usage where constraint_name = ?" constraint-name]
-       (jdbc/query conn {:identifers identity, :transaction? false})
-       (map :column_name)))
-
-(defn- violates-not-null-constraint [_conn error-message]
+(defn- maybe-parse-not-null-error [_database error-message]
```

Comment:
I couldn't help myself and renamed these functions to better match what they actually **do**.